templates/default/consul-template-init.erb

```diff
@@ -1,109 +1,77 @@
-#                    Consul into the filesystem using the consul-template daemon.
-### END INIT INFO
-
-CMD="<%= @command %> <%= @options %>"
+#!/bin/bash
+PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin
 NAME="consul-template"
 
-PIDFILE="/var/run/$NAME.pid"
+USER=root
+#DAEMON="<%= @command %>"
+DAEMON="/usr/local/bin/consul-template"
+
 LOGFILE="/var/log/$NAME.log"
 
-[ -f /etc/default/${NAME} ] && . /etc/default/${NAME}
-export GOMAXPROCS=${GOMAXPROCS}
+SCRIPTNAME=/etc/init.d/${NAME}
+PIDFILE="/var/run/$NAME.pid"
+LOCKFILE="/var/run/lock/${NAME}"
+SHUTDOWN_TIMEOUT=1
 
-get_pid() {
-    cat "$PIDFILE"
-}
+# DEAMON_OPTS="<%= @options %>"
+DEAMON_OPTS="-config /etc/consul-template/conf.d"
 
-is_running() {
-    [ -f "$PIDFILE" ] && ps `get_pid` > /dev/null 2>&1
-}
+export GOMAXPROCS=${GOMAXPROCS}
 
-case "$1" in
-    start)
-    if is_running; then
-        echo "$NAME already running"
-    else
-        echo "Starting $NAME"
-        CONSUL_TEMPLATE_LOG=<%= @loglevel %> $CMD >> "$LOGFILE" 2>&1 &
-        echo $! > "$PIDFILE"
-        if ! is_running; then
-            echo "Unable to start $NAME, see $LOGFILE"
-            exit 1
-        fi
-    fi
-    ;;
-    stop)
-    if is_running; then
-        echo -n "Stopping $NAME..."
-        kill `get_pid`
-        for i in 1 2 3 4 5 6 7 8 9 10
-        do
-            if ! is_running; then
-                break
-            fi
+. /lib/lsb/init-functions
 
-            echo -n "."
-            sleep 1
-        done
-        echo
+start() {
+        echo "Starting $NAME: "
+        daemonize -u $USER -p $PIDFILE -l $LOCKFILE -o $LOGFILE -e $LOGFILE -E GOMAXPROCS=${GOMAXPROCS} $DAEMON $DEAMON_OPTS
+        log_end_msg $?
+}
 
-        if is_running; then
-            echo "$NAME not stopped; may still be shutting down or shutdown may have failed"
-            exit 1
-        else
-            echo "$NAME stopped"
-            if [ -f "$PIDFILE" ]; then
-                rm "$PIDFILE"
-            fi
+stop(){
+        [ -e "$PIDFILE" ] && [ ! -z "$(cat $PIDFILE)" ] && test -d "/proc/$(cat $PIDFILE)" && kill -SIGTERM $(cat $PIDFILE)
+        log_daemon_msg "Stopping $NAME: "
+        flock --exclusive --timeout $SHUTDOWN_TIMEOUT $LOCKFILE --command "echo -n"
+        if [ $? -ne 0 ]; then
+                kill -SIGKILL $(cat $PIDFILE)
+                flock --exclusive --timeout 1 $LOCKFILE --command "echo -n"
         fi
-    else
-        echo "$NAME not running"
-    fi
-    ;;
-    restart)
-    $0 stop
-    if is_running; then
-        echo "Unable to stop $NAME, will not attempt to start"
-        exit 1
-    fi
-    $0 start
-    ;;
-    status)
-    if is_running; then
-        echo "$NAME is running"
-    else
+        log_end_msg $?
+}
+reload(){
+        [ -e "$PIDFILE" ] && [ ! -z "$(cat $PIDFILE)" ] && test -d "/proc/$(cat $PIDFILE)" && kill -SIGHUP $(cat $PIDFILE)
+        sleep 1
+        log_daemon_msg "reloading $NAME: "
+        log_end_msg $?
```

Comment:
This is not going to work as intended because $? returns an exit cde of the last command